src/testModel.ts

```diff
@@ -665,7 +690,10 @@ export class TestModelCollection extends DisposableBase {
     const configSettings = (workspaceSettings.configs || []).find(c => c.relativeConfigFile === path.relative(model.config.workspaceFolder, model.config.configFile));
     model.isEnabled = configSettings?.enabled || (this._models.length === 1 && !configSettings);
     await this._loadModelIfNeeded(model);
-    this._disposables.push(model.onUpdated(() => this._didUpdate.fire()));
+    this._disposables.push(
+        model,
```

Comment:
`TestModelCollection.clear()` will call both dispose() and reset() now. Also you might want to hide model's trace viewer when model is being reset (that happens when it is enabled and disabled, it is worth closing trace viewer on that signal.

Reply:
> TestModelCollection.clear() will call both dispose() and reset() now

You're right, I'll change it to ensure it only calls `testModel.reset`, otherwise we'll end up with some nasty bugs.

> Also you might want to hide model's trace viewer when model is being reset

Trace viewer already closes when model is reset:

https://github.com/microsoft/playwright-vscode/blob/f01b9d47487d2d93091d8305aa5912e28455445d/src/testModel.ts#L124

However, it doesn't close when another test model is selected. I'll implement that.

Reply:
Actually,, `TestModelCollection.clear()` actually disposes testModels, so I think the current behaviour is right. If it only called `reset`, `options.settingsModel.showTrace.onChange` event handler would not be disposed.

Reply:
Ok, sounds good. We should probably follow up with getting rid of

```js
clear() {
  this.dispose();
}
```

for clarity. `clear` assumes that the instance will be reused. But if in the future someone adds an essential listener into `disposables` in TestModelCollection, calling `clear()` will be destroying it and hence leading to bugs in TestModelCollection.

Reply:
Maybe to make it clear:

- instead of extend `DisposableBase`, just implement `vscodeType.Disposable`
- intead of `_disposables`, keep model disposables in a `private _modelsDisposables: Disposable[];`
- implement `clear` and `dispose` functions this way:

```js
  clear() {
    this._disposeModels();
    this._didUpdate.fire();
  }

  dispose() {
    this._disposeModels();
  }

  private _disposeModels() {
    for (const disposable of this.__modelsDisposables)
     disposable.dispose();
    this._models = [];
  }
```

Reply:
Actually, to move the `onUpdated` inside of `TestModelCollection` we still need to keep the `DisposableBase` and keep the additional `_modelsDisposables` property to release cleared models. I applied those changes in b8551f60a0b309ca1627399fe060490ddcb910d2